tests/phpunit/tests/test-install.php

```diff
@@ -67,19 +86,10 @@ function test_uninstall() {
 
 		update_post_meta( $item_id, '_pll_menu_item', array() );
 
-		$this->assertFalse( defined( 'PLL_REMOVE_ALL_DATA' ), 'The constant PLL_REMOVE_ALL_DATA must not be defined for this test'  );
-
 		if ( ! defined( 'WP_UNINSTALL_PLUGIN' ) ) {
```

Comment:
Since you defined it in `PLL_Install_Test::test_uninstall()`, no need to do it here. Besides, you already need `PLL_Install_Test::test_uninstall()` to be executed before this one, because the `PLL_REMOVE_ALL_DATA` constant set in this code would break the test if they were performed in the other order.

It seems that test are performed in sequential order in the class they're written on. Otherwise, you might use the [@depends](https://phpunit.de/manual/5.7/en/appendixes.annotations.html#appendixes.annotations.depends) annotation if tests needs to be played in a specific order (beware, if the dependency test fails, the dependent test would be skipped).

Reply:
The 2 tests are not really dependent on each other. So I prefer to keep theme completely separated.